src/common/framework/gpu/device_pool.ts

```diff
@@ -35,6 +35,31 @@ export class DevicePool {
     return this.holder.device;
   }
 
+  async acquireWithDeviceDescriptor(
+    devicedescriptor: GPUDeviceDescriptor
+  ): Promise<GPUDevice | null> {
+    assert(!this.failed, 'WebGPU device previously failed to initialize; not retrying');
+
+    if (this.holder === undefined) {
+      try {
+        this.holder = await DevicePool.makeHolderWithDeviceDescriptor(devicedescriptor);
```

Comment:
This logic doesn't exactly make sense, because the devicedescriptor will be ignored if there was ever a device created before. For this to be robust, the device descriptor should simply be supplied in the DevicePool constructor.

Ultimately I think we want to key the pool off of the device creation options (i.e. have a map of descriptor -> pool). But I haven't figured out how to implement that exactly. I was thinking about getting to this soon so since you have started writing extension tests I might look at it today.

Reply:
Thanks for pointing out this issue! I've tried to improve it a bit more. PTAL, thanks!